File: Lab.SignalR.Server/Views/GoodsReceipts/GoodsReceipt.ts
export class GoodsReceipt {

    constructor(public id : number, public positions : GoodsReceiptPosition[], comments: string[]) {}

}

export class GoodsReceiptPosition {
    constructor(public articleNumber : string, public articleDescription : string, public orderedAmount : number, public devliveredAmount : number) {}
}

File: Dnp.SignalR.Server/Models/GoodsReceipt.cs
using System.Collections.Generic;

namespace Dnp.SignalR.Server.Models
{

    public class GoodsReceipt
    {

        public GoodsReceipt()
        {
            Positions = new List<GoodsReceiptPosition>();
            Comments = new List<string>();
        }

        public int Id { get; set; }

        public List<GoodsReceiptPosition> Positions { get; set; }

        public List<string> Comments { get; set; }

    }

    public class GoodsReceiptPosition
    {

        public string ArticleNumber { get; set; }

        public string ArticleDescription { get; set; }

        public int OrderedAmount { get; set; }

        public int DeliveredAmount { get; set; }


    }
}